src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx

```diff
@@ -73,10 +71,11 @@ export const WalletSelectionScreen = () => {
       await showErrorDialog(errorMessages.itnNotSupported, intl)
       return
     }
-    if (wallet?.id === walletMeta.id) {
-      return navigateToTxHistory()
+    if (params?.reopen || wallet?.id !== walletMeta.id) {
```

Comment:
whats going on here?

Reply:
we should avoid using params

Reply:
The optimization to just navigate to the wallet history when the wallet selected is the current one needs to be avoided when the user resync the wallet.

Reply:
About the params, my idea is to tackle all params as feature context in a ticket for it.